diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SortRule.kt

```diff
@@ -80,12 +82,11 @@ class SortRule(private val configRules: List<RulesConfig>) : Rule("sort-rule") {
         sortList: List<ASTNode>,
         nonSortList: List<ASTNode>,
         node: ASTNode) {
+        val spaceBefore = nonSortList.map { astNode -> astNode.prevSibling { it.elementType == WHITE_SPACE } }
         val nodeBefore: ASTNode? = nonSortList.last().treeNext
         node.removeRange(nonSortList.first(), nonSortList.last().treeNext)
         sortList.forEachIndexed { nodeIndex, astNode ->
-            if (nodeIndex != 0) {
-                node.addChild(PsiWhiteSpaceImpl("\n"), nodeBefore)
-            }
+            spaceBefore[nodeIndex]?.let { node.addChild(it, nodeBefore) }
```

Comment:
OK, so now you keep spaces when swapping enum members. But how did it lead to the error described in the issue? Is it something to do with enum members without spaces between them (`A,B` instead of `A, B`) or is something with interaction between this rule and `WhitespaceRule` or `NewlinesRule`? Also, what happens if input contains members without space after comma?

Reply:
The error was in the `swapSortNodes` method. In our algorithm, after each insertion of the correct element, we also insert a new line

Reply:
Yes, I see now, OK.